src/main/java/software/amazon/nio/spi/s3/S3TransferException.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.nio.spi.s3;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+
+public class S3TransferException extends IOException {
+    private static final long serialVersionUID = 1L;
+
+    private final String errorCode;
+    private final String errorMessage;
+    private final Integer numAttempts;
+    private final String requestId;
+    private final int statusCode;
+
+    public S3TransferException(String method, Path path, AwsServiceException cause) {
+        this(
+            errorCode(cause),
+            errorMessage(cause),
+            cause.requestId(),
+            cause.statusCode(),
+            cause.numAttempts(),
+            message(method, path, cause),
+            cause);
+    }
+
+    private S3TransferException(
+        String errorCode,
+        String errorMessage,
+        String requestId,
+        int statusCode,
+        Integer numAttempts,
+        String message,
+        Throwable cause) {
+        super(message, cause);
+        this.errorCode = errorCode;
+        this.errorMessage = errorMessage;
+        this.requestId = requestId;
+        this.statusCode = statusCode;
+        this.numAttempts = numAttempts;
+    }
+
+    private static String errorCode(AwsServiceException cause) {
```

Comment:
These `private static` methods appear to be unused. What are they intended for? Can they be removed?

Reply:
The method `errorCode` is used on line 38.